internal/stats/metrics_recorder_list.go

```diff
@@ -103,3 +103,22 @@ func (l *MetricsRecorderList) RecordInt64Gauge(handle *estats.Int64GaugeHandle,
 		metricRecorder.RecordInt64Gauge(handle, incr, labels...)
 	}
 }
+
+// NoopMetricsRecorder is a noop MetricsRecorder to be used to prevent nil
+// panics.
+type NoopMetricsRecorder struct{}
```

Comment:
I think you can just use an empty MetricsRecorderList instead?

Reply:
Yeah I didn't like that since it's a top level object of channel and server so it felt weird. I'll delete this and switch it to that though to not have to add this type.